nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -152,88 +144,126 @@ def run_conversion(
             f"({len(image_series_kwargs_list)}) vs. file_paths ({len(self.source_data['file_paths'])})!"
         )
 
-        for j, file in enumerate(file_paths):
-            timestamps = starting_times[j] + get_movie_timestamps(movie_file=file)
-
-            if len(starting_times) != len(file_paths):
-                starting_times.append(timestamps[-1])
+        def _check_duplicates(image_series_kwargs_list):
+            image_series_kwargs_list_keys = [i["name"] for i in image_series_kwargs_list]
+            if len(set(image_series_kwargs_list_keys)) < len(image_series_kwargs_list_keys):
+                assert external_mode, "For multiple video files under the same ImageSeries name, use exernal_mode=True."
+            keys_set = []
+            image_series_kwargs_list_unique = []
+            file_paths_list = []
+            for n, image_series_kwargs in enumerate(image_series_kwargs_list):
+                if image_series_kwargs["name"] not in keys_set:
+                    keys_set.append(image_series_kwargs["name"])
+                    file_paths_list.append([file_paths[n]])
+                    image_series_kwargs_list_unique.append(dict(image_series_kwargs))
+                else:
+                    idx = keys_set.index(image_series_kwargs["name"])
+                    file_paths_list[idx].append(file_paths[n])
+            return image_series_kwargs_list_unique, file_paths_list
 
-            image_series_kwargs = dict(image_series_kwargs_list[j])
-            if check_regular_timestamps(ts=timestamps):
-                fps = get_movie_fps(movie_file=file)
-                image_series_kwargs.update(starting_time=starting_times[j], rate=fps)
-            else:
-                image_series_kwargs.update(timestamps=H5DataIO(timestamps, compression="gzip"))
+        image_series_kwargs_list_updated, file_paths_list = _check_duplicates(image_series_kwargs_list)
+        if starting_times is not None:
+            assert isinstance(starting_times, list) and all(
+                [isinstance(x, float) for x in starting_times]
+            ), "Argument 'starting_times' must be a list of floats."
+            assert len(starting_times) == len(image_series_kwargs_list_updated), (
+                "starting times list length must be equal number of unique ImageSeries " "containers to write to nwb"
+            )
+        else:
+            warn("starting_times not provided, setting to 'None'")
+            starting_times = [None] * len(image_series_kwargs_list_updated)
```

Comment:
```suggestion
            starting_times = [np.nan] * len(image_series_kwargs_list_updated)
```

What you said earlier makes more sense; `NaN` would be better